cmd/korrel8r/main.go

```diff
@@ -88,6 +91,14 @@ func main() {
 		}
 		os.Exit(0)
 	}()
+	// Extra settings for REST configuration used by most stores and k8s clients.
+	k8s.ConfigureConfig = func(cfg *rest.Config) {
+		// TODO make these configurable. Use viper?
+		cfg.QPS = 100
+		cfg.Burst = 1000
+		cfg.Timeout = 5 * time.Second
+		cfg.Wrap(auth.Wrap)
```

Comment:
I think there's no need to go the roundtripper-route for the authentication. The `rest.Config` struct has an attribute for setting the bearer-token to use (`BearerToken`) along with a helper function to remove all other authentication options (`rest.AnonymousClientConfig()`):

```
cfg = rest.AnonymousClientConfig(cfg)
cfg.BearerToken = token
```

Note: The `token` here is the bare token, not the whole `Authorization` header.

Reply:
Injecting a roundtripper into the common rest.Config means every store automatically gets the auth behavior, if the Context contains an auth hint. Since every store currently uses the common rest.Config, I don't have to touch anything else. I could be more specific about bearer tokens (currently this just copies any Authorization header) I'll note that for future.

Reply:
We might have talked past each other. This still uses `rest.Config` as before, just removes all the authentication methods from it (which might not just be bearer-token) and sets the specific field for bearer-token authentication in the client. This removes the need for a custom `RoundTripper` implementation but the "one `rest.Config` for all stores" is still the same.